vector/src/main/java/im/vector/app/features/settings/devices/v2/DevicesViewModel.kt

```diff
@@ -116,4 +138,80 @@ class DevicesViewModel @AssistedInject constructor(
     private fun handleMarkAsManuallyVerifiedAction() {
         // TODO implement when needed
     }
+
+    private fun handleMultiSignoutOtherSessions() = withState { state ->
+        viewModelScope.launch {
+            setLoading(true)
+            val deviceIds = getDeviceIdsOfOtherSessions(state)
+            if (deviceIds.isEmpty()) {
+                return@launch
+            }
+            val signoutResult = signout(deviceIds)
+            setLoading(false)
+
+            if (signoutResult.isSuccess) {
+                onSignoutSuccess()
+            } else {
+                when (val failure = signoutResult.exceptionOrNull()) {
+                    null -> onSignoutSuccess()
```

Comment:
In fact, when `SignoutResult.exceptionOrNull()` returns null it means it is of type `Success`. I had to handle this case but technically this branch cannot be reached here since we check the type of the result before.

Reply:
I updated the code to avoid this nonsense. See [6d26208](https://github.com/vector-im/element-android/pull/7456/commits/6d2620815cc3923759b3d17690aaaf15eb1f26d6)